File: apps/pos/native-android/app/src/main/java/com/desktopkitchen/pos/utilities/CurrencyFormatter.kt
package com.desktopkitchen.pos.utilities

import java.text.NumberFormat
import java.util.Locale

object CurrencyFormatter {
    const val TAX_RATE = 0.16
    const val TAX_LABEL = "IVA (16%)"

    private val formatter = NumberFormat.getCurrencyInstance(Locale("es", "MX")).apply {
        minimumFractionDigits = 2
        maximumFractionDigits = 2
    }

    fun format(amount: Double): String = formatter.format(amount)

    fun formatShort(amount: Double): String = String.format(Locale.US, "$%.2f", amount)

    /** Extract IVA from a tax-inclusive total (Mexican pricing). */
    fun extractTax(fromTotal: Double): Double {
        val raw = fromTotal - fromTotal / (1 + TAX_RATE)
        return Math.round(raw * 100.0) / 100.0
    }

    /** Extract the pre-tax subtotal from a tax-inclusive total. */
    fun extractSubtotal(fromTotal: Double): Double {
        val raw = fromTotal / (1 + TAX_RATE)
        return Math.round(raw * 100.0) / 100.0
    }
}

File: apps/pos/native-windows/DesktopKitchenPOS/Utilities/CurrencyFormatter.cs
using System.Globalization;

namespace DesktopKitchenPOS.Utilities;

public static class CurrencyFormatter
{
    public const double TaxRate = 0.16;
    public const string TaxLabel = "IVA (16%)";

    private static readonly CultureInfo MxCulture = new("es-MX");

    public static string Format(double amount)
    {
        return amount.ToString("C2", MxCulture);
    }

    public static string FormatShort(double amount)
    {
        return $"${amount:F2}";
    }

    /// <summary>
    /// Extract IVA from a tax-inclusive total (Mexican pricing).
    /// </summary>
    public static double ExtractTax(double fromTotal)
    {
        return Math.Round(fromTotal - fromTotal / (1 + TaxRate), 2);
    }

    /// <summary>
    /// Extract the pre-tax subtotal from a tax-inclusive total.
    /// </summary>
    public static double ExtractSubtotal(double fromTotal)
    {
        return Math.Round(fromTotal / (1 + TaxRate), 2);
    }
}
